### ISSUE TITLE
Visio error when using page.activate and a load() method together

### ISSUE THREAD

**Issue by mblennegard** [OfficeDev/office-js#170]
## Expected Behavior
I am trying to retrieve the first page of a Visio chart, using the Visio Online API, first by retrieving the page name, which is then used to activate the specific page.

The expected behavior is that the page will be activated.

## Current Behavior
When using, what appears to be any load() method, in the same session (OfficeExtension.EmbeddedSession), the Visio throws an error directly in the chart (as opposed to a console error).

![image](https://user-images.githubusercontent.com/27766499/41532319-d69a2bce-72f6-11e8-9ac2-2dac0075812b.png)

## Steps to Reproduce, or Live Example
The issue is produced by using the following code, with the API file located here: https://appsforoffice.microsoft.com/embedded/1.0/visio-web-embedded.js.

``` javascript
Visio.run(session, function (vctx) {
    var pages = vctx.document.pages;
    var document = vctx.document;
    
    var pageName;
    pages.load();

    return vctx.sync().then(function () {
        pageName = pages.items[0].name;
        var page = pages.getItem(pageName);
        page.activate();
        console.log("Page name retrieved")

        return vctx.sync().then(function () {
            console.log("Page activated")
        });
    });
}).catch(function (error) {
    console.log("Error: " + error);
    if (error instanceof OfficeExtension.Error) {
        console.log("Debug info: " + JSON.stringify(error.debugInfo));
    }
});
```

As soon as I comment out the row **pages.load();**, everything works fine (by manually setting the pageName. However, if leaving the load() method, even though the pageName is still set manually, the error is thrown, so the issue seems to only occur when mixing the **load()** and **activate()** methods within the same session or similar).
I have tried with both pages.load() and page.load() methods, both of which throw the error.

Besides the page.activate() method, there is also the document.setActivePage() method, which causes the exact same error, when combined with a load() method.

## Context
I need to retrieve the first page name in the Visio chart, which is then being used in the next call for activating that specific page.

## Your Environment
* Platform [PC desktop, Mac, iOS, Office Online]: PC, SharePoint Online
* Operating System: Windows 10
* Browser (if using Office Online): Chrome, Edge, Internet Explorer 11

**Comment by rahul-mi** [comment]
I believe this was due to another known issue, which has been fixed now and should be available for you. load() and activate() can be used together and I tried out your code and it executed fine. Can you confirm if it works for you now or you are still facing issues?

On a related note, you can also activate that page by using document.setActivePage(pageName) API which removes the dependency on activate() API.

```js
Visio.run(session, function (vctx) {
    var pages = vctx.document.pages;
    var document = vctx.document;
    
    var pageName;
    pages.load();

    return vctx.sync().then(function () {
        pageName = pages.items[0].name;
        document.setActivePage(pageName);
        console.log("Page name activated")
    });
}).catch(function (error) {
    console.log("Error: " + error);
    if (error instanceof OfficeExtension.Error) {
        console.log("Debug info: " + JSON.stringify(error.debugInfo));
    }
});
```
You can also checkout some working code examples for Visio Online APIs, available in the blog [here](https://blogs.msdn.microsoft.com/visio/2017/09/25/migrate-from-visio-web-access-to-visio-online/) . 
(Section - Embed a Visio drawing in a classic SharePoint page with key properties).

**Comment by mblennegard** [comment]
Hi,

The issue still persists, even when using the document.setActivePage()
method.

I am using the following API file:
https://appsforoffice.microsoft.com/embedded/1.0/visio-web-embedded.js">

I believe this is the latest version of the API?

Best regards,
Marcus

Den fre 22 juni 2018 kl 15:41 skrev Rahul Mittal <notifications@github.com>:

**Comment by rahul-mi** [comment]
Hi,

I tried out your code in Edge browser and found it working, so I believed that the issue might have been due to something else that we fixed recently. 

I deleted the comment later as I found that issue still exists on Chrome. Let us investigate the issue and come back. Thank you for your patience.

And yes, the above link points to the latest version of the API.

Regards,
Rahul Mittal

**Comment by mblennegard** [comment]
Hello again,

Now when I try it, it works fine in both Chrome and Internet Explorer.
Have you done something new to the API file?

Last night when I tried it didn't work.

Best regards,
Marcus

Den mån 25 juni 2018 kl 06:32 skrev Rahul Mittal <notifications@github.com>:

**Comment by mblennegard** [comment]
Hmm, when trying now it doesn't work again.
Seems like the issue was temporarily gone, but is now back again.

**Comment by mblennegard** [comment]
I have kept testing for the last few days and it seems that this error has been fixed.

My function now works perfectly, first retrieving the first page name after the **pages.load()** method has been executed via a **context.sync** call and then activated via the **page.activate()** method.
The document.setActivePage() method works fine as well.

However, the original error (in my first post for this issue), now occurs when performing another context.sync after my function has been executed once.
If I try calling my function another time, I get a console.log message that something is undefined when performing a context.sync.
The error message I receive, when calling my function again, is the following.
![image](https://user-images.githubusercontent.com/27766499/42023611-94b40528-7ac0-11e8-8770-9cac82b416a9.png)

I have checked all my own variables and the first page name is still retrieved fine.
Somehow it seems to throw errors when performing a context.sync, in another Visio.run, after a Visio.run has already been performed, all the time connected somehow to the original problem (as I can run several Visio.run when not using my function above with activate()).
